client/dashboard/task-list/tasks/appearance.js

```diff
@@ -25,20 +26,27 @@ class Appearance extends Component {
 		super( props );
 
 		this.state = {
+			isPending: false,
 			logo: null,
 			stepIndex: 0,
+			steps: [],
 			storeNoticeText: props.options.woocommerce_demo_store_notice || '',
 		};
 
 		this.completeStep = this.completeStep.bind( this );
+		this.importProducts = this.importProducts.bind( this );
 		this.updateLogo = this.updateLogo.bind( this );
 		this.updateNotice = this.updateNotice.bind( this );
 	}
 
+	componentWillMount() {
```

Comment:
You're right.  This was also problematic since `isPending` or other state items wouldn't be updated here.  Moved the visibility to the constructor instead to keep those static per instance.